gems/sorbet-runtime/lib/types/struct.rb

```diff
@@ -2,9 +2,19 @@
 # typed: true
 
 class T::InexactStruct
+  extend T::Sig
+
   include T::Props
   include T::Props::Serializable
   include T::Props::Constructor
+
+  sig {params(other: Object).returns(T::Boolean)}
+  def ==(other)
+    return true if self.equal?(other)
+    return false unless other.is_a?(self.class)
+
+    self.class.props.all? { |prop, _| self.send(prop) == other.send(prop) }
```

Comment:
Let's not use `send` and iterate over prop values directly

Reply:
As in via `#instance_variable_get`? Would that be faster? Or are you proposing something more along the lines of `serialize`'s source is generated and eval-ed?